bin/fetch-configlet.ps1

```diff
@@ -2,25 +2,28 @@ $ErrorActionPreference = "Stop"
 $ProgressPreference = "SilentlyContinue"
 
 $requestOpts = @{
-    Headers = If ($env:GITHUB_TOKEN) { @{ Authorization = "Bearer ${env:GITHUB_TOKEN}" } } Else { @{ } }
+    Headers           = If ($env:GITHUB_TOKEN) { @{ Authorization = "Bearer ${env:GITHUB_TOKEN}" } } Else { @{ } }
     MaximumRetryCount = 3
-    RetryIntervalSec = 1
+    RetryIntervalSec  = 1
 }
 
-$arch = If ([Environment]::Is64BitOperatingSystem) { "64bit" } Else { "32bit" }
-$fileName = "configlet-windows-$arch.zip"
-
-Function Get-DownloadUrl {
+function Get-DownloadUrl($FileName, $RequestOpts) {
+    if ($PSVersionTable.PSVersion.Major -le 5) {
+        # PreserveAuthorizationOnRedirect requires PowerShell 7+
+        throw "Please upgrade to PowerShell Core. For example: 'dotnet tool install --global PowerShell'"
+    }
+    
     $latestUrl = "https://api.github.com/repos/exercism/configlet/releases/latest"
-    Invoke-RestMethod -Uri $latestUrl -PreserveAuthorizationOnRedirect @requestOpts
-    | Select-Object -ExpandProperty assets
-    | Where-Object { $_.browser_download_url -match $FileName }
-    | Select-Object -ExpandProperty browser_download_url
+    $assets = Invoke-RestMethod -Uri $latestUrl -PreserveAuthorizationOnRedirect @RequestOpts | Select-Object -ExpandProperty assets
+    $assets | Where-Object { $_.browser_download_url -match $FileName } | Select-Object -ExpandProperty browser_download_url
 }
 
-$downloadUrl = Get-DownloadUrl
 $outputDirectory = "bin"
+$arch = If ([Environment]::Is64BitOperatingSystem) { "64bit" } Else { "32bit" }
+$fileName = "configlet-windows-$arch.zip"
+
+$downloadUrl = Get-DownloadUrl $fileName $requestOpts
 $outputFile = Join-Path -Path $outputDirectory -ChildPath $fileName
-Invoke-WebRequest -Uri $downloadUrl -OutFile $outputFile @requestOpts
+Invoke-WebRequest -Uri $downloadUrl -OutFile $outputFile -PreserveAuthorizationOnRedirect @requestOpts
```

Comment:
I think I purposefully not added that as it did not work. Did you check to see if it worked?

Reply:
Both with and without works fine for me, so wanted to use only named parameters.

Reply:
Ah, it does indeed work on Windows. I tested it using powershell on Linux (WSL), where it fails with:

```
erik@VELPIE:~/csharp$ pwsh ./bin/fetch-configlet.ps1 
Invoke-WebRequest: /home/erik/csharp/bin/fetch-configlet.ps1:27:1
Line |
  27 |  Invoke-WebRequest -Uri $downloadUrl -OutFile $outputFile -PreserveAut …
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Cannot access a disposed object. Object name: 'System.Net.Http.HttpClient'.
```

This is apparently some bug that only occurs on linux. I'd really like the powershell to also work on linux, so could you revert this change?

Reply:
I tested on WSL ubuntu 20.04.1 LTS and WSL 18.04LTS with pwsh 7.1 on Win10-20H2, and both with and without the named parameter `-PreserveAuthorizationOnRedirect`  work fine.

What was you tested on?